keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,832 @@
+  // -  GCS: Indicates Google Cloud Storage protocol
+  Protocol Protocol
+
+  // DeletionPolicy is used to specify how COSI should handle deletion of this
+  // bucket. There are 3 possible values:
+  //  - Retain: Indicates that the bucket should not be deleted from the OSP
+  //  - Delete: Indicates that the bucket should be deleted from the OSP
+  //        once all the workloads accessing this bucket are done
+  DeletionPolicy DeletionPolicy
+
+  // Parameters is an opaque map for passing in configuration to a driver
+  // for creating the bucket
+  // +optional
+  Parameters map[string]string
 }
 ```
 
-## ProvisonerCreateBucket
+## BucketAccess
 
-This call is made to create the bucket in the backend. If a bucket that matches both name and parameters already exists, then OK (success) must be returned. If a bucket by same name, but different parameters is provided, then the appropriate error code `ALREADY_EXISTS` must be returned by the provisioner. The call to *ProvisonerCreateBucket* MUST be idempotent.
+A resource to access a Bucket. BucketAccess is namespace-scoped
 
+```yaml
+BucketAccess {
+  TypeMeta
+  ObjectMeta
+
+  Spec BucketClaimSpec {
+    // BucketClaimName is the name of the BucketClaim.
+    // Exactly one of BucketClaimName or BucketName must be set.
+    // +optional
+    BucketClaimName string
+
+    // BucketName is the name of the Bucket for which
+    // credetials need to be generated
+    // Exactly one of BucketClaimName or BucketName must be set.
+    // +optional
+    BucketName string
+
+    // BucketAccessClassName is the name of the BucketAccessClass
+    BucketAccessClassName string
+
+    // CredentialsSecretName is the name of the secret that COSI should populate
+    // with the credentials. If a secret by this name already exists, then it is
+    // assumed that credentials have already been generated. It is not overridden.
+    CredentialsSecretName string
+    
+    // ServiceAccountName is the name of the serviceAccount that COSI will map
+    // to the OSP service account when IAM styled authentication is specified
+    CredentialsSecretName string
+  }
+
+  Status BucketAccessStatus {
+    // AccessGranted indicates the successful grant of privileges to access the bucket
+    AccessGranted bool
+    
+    // AccountID is the unique ID for the account in the OSP
+    AccountID string
+  }
 ```
-message ProvisionerCreateBucketRequest {    
-    // This field is REQUIRED
-    string bucket_name = 1;
 
-    map<string,string> bucket_context = 2;
+## BucketAccessClass
 
-    enum AnonymousBucketAccessMode {
-	PRIVATE = 0;
-	PUBLIC_READ_ONLY = 1;
-	PUBLIC_WRITE_ONLY = 2;
-	PUBLIC_READ_WRITE = 3;
-    }
-    
-    AnonymousBucketAccessMode anonymous_bucket_access_mode = 3;
-}
+Resouce for configuring common properties for multiple BucketClaims. BucketAccessClass is a clustered resource
 
-message ProvisionerCreateBucketResponse {
-    // Intentionally left blank
+```yaml
```

Comment:
Does this need a Driver field?

Reply:
Yes, updated kep to indicate a driver field in BucketAccessClass

Reply:
This seems error prone. What if a user creates a Bucket using provisioner A and then references a BucketAccessClass that is only supported with provisioner B?

Reply:
That's a user error. Since, we do not support buckets created by provisioner A to be managed by provisioner B, the same would apply for BucketAccess too.

Reply:
How does a user figure out which BucketAccessClasses are valid ones for the BucketClass that they have chosen?

Reply:
I think for discoverability it's nicer to have the Driver name in the BucketAccessClass so a user can easily filter and see what is available to them.

Reply:
still open

Reply:
done